src/buildercore/cfngen.py

```diff
@@ -253,18 +253,27 @@ def build_subdomain(x):
         context['cloudfront'] = False
 
 def build_context_fastly(context, parameterize):
-    def build_subdomain(x):
+    def _build_subdomain(x):
         return complete_domain(parameterize(x), context['domain'])
+
+    def _parameterize_hostname(b):
+        b['hostname'] = parameterize(b['hostname'])
+        return b
+
     if 'fastly' in context['project']['aws']:
+        backends = context['project']['aws']['fastly'].get('backends', OrderedDict({}))
         context['fastly'] = {
-            'subdomains': [build_subdomain(x) for x in context['project']['aws']['fastly']['subdomains']],
-            'subdomains-without-dns': [build_subdomain(x) for x in context['project']['aws']['fastly']['subdomains-without-dns']],
+            'backends': OrderedDict([(n, _parameterize_hostname(b)) for n, b in backends.items()]),
```

Comment:
`.items()` won't have a guaranteed order (unless `backends` is also an `OrderedDict`?)

Reply:
It is https://github.com/elifesciences/builder/pull/321/files/de7dc59419d52adeb6daa9006fdf396f2c31571e#diff-11aa22c619b41fcc4ac5151f83decddcR264
Was causing a random test failure, which then was resolved